File: src/main/java/factorio/debugger/DAP/messages/types/DAPSteppingGranularity.java
package factorio.debugger.DAP.messages.types;

import com.fasterxml.jackson.annotation.JsonProperty;

public enum DAPSteppingGranularity {
    @JsonProperty("statement")
    STATEMENT,
    @JsonProperty("line")
    LINE,
    @JsonProperty("instruction")
    INSTRUCTION
}

File: src/main/kotlin/factorio/debugger/DAP/messages/types/DAPSteppingGranularity.kt
package factorio.debugger.DAP.messages.types

import com.fasterxml.jackson.annotation.JsonProperty

enum class DAPSteppingGranularity {
    @JsonProperty("statement")
    STATEMENT,
    @JsonProperty("line")
    LINE,
    @JsonProperty("instruction")
    INSTRUCTION;

    override fun toString(): String {
        return this.name.lowercase()
    }
}
